--- src/interp/ncomp.boot	2014-03-12 23:25:51.000000000 +0100
+++ ncomp.boot	2014-04-22 15:03:28.000000000 +0200
@@ -325,6 +325,10 @@
 DEFVAR($globalDefs, nil)
 DEFVAR($MacroTable)
 
+DEFVAR($PostParseHookEnable, false)
+DEFVAR($PostParseHookFunction, nil)
+DEFVAR($PostParseHookDomain, nil)
+
 S_process(x) ==
     $Index : local := 0
     $MACROASSOC : local := nil
@@ -369,6 +373,14 @@
     $PrintOnly =>
         FORMAT(true, '"~S   =====>~%", $currentLine)
         PRETTYPRINT(x)
+    if $PostParseHookEnable then
+        fn := getFunctionFromDomain($PostParseHookFunction, $PostParseHookDomain, [])
+        FORMAT(true, '"~%~A~%~%", '">>>> POST-PARSE HOOK BEGIN <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<")
+        if null(fn) then
+            SAY("Post-parse hook not found!")
+        else
+            SPADCALL(x, fn)
+        FORMAT(true, '"~%~A~%~%", '">>>> POST-PARSE HOOK END <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<")
     u := compTopLevel(x, $EmptyMode, $InteractiveFrame)
     if u then $InteractiveFrame := THIRD(u)
     if $semanticErrorStack then displaySemanticErrors()
